drawing_agent/main.py

```
@@ -61,133 +31,168 @@ class SearchRequest(BaseModel):
+        query_embedding = agent_instance.drawing_knowledge.embed_model.generate(clean_query)
+
+        results = agent_instance.vector_db.search(
+            query_embedding=query_embedding,
+            drawing_id=drawing_id,
+            k=req.limit
+        )
+
+        return {"success": True, "results": results}
+    except Exception as e:
+        logger.exception("Search failed")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)